deviations/views.py

```diff
@@ -2,13 +2,59 @@
 from django.db import IntegrityError
 from django.shortcuts import get_object_or_404
 from django.utils.translation import ugettext_lazy as _
+from userprofile.models import UserProfile
 
 from course.viewbase import CourseInstanceBaseView, CourseInstanceMixin
 from lib.viewbase import BaseFormView, BaseRedirectView
 from authorization.permissions import ACCESS
-from .forms import DeadlineRuleDeviationForm
+from .forms import DeadlineRuleDeviationForm, RemoveDeadlineRuleDeviationForm
 from .models import DeadlineRuleDeviation
 from exercise.models import BaseExercise
+from itertools import chain
+
+
+class OverrideDeadlinesView(CourseInstanceMixin, BaseFormView):
+    access_mode = ACCESS.TEACHER
+    template_name = "deviations/override.html"
+    form_class = DeadlineRuleDeviationForm
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["instance"] = self.instance
+        return kwargs
+
+    def get_success_url(self):
+        return self.instance.get_url("deviations-list-dl")
+
+    def get_common_objects(self):
+        super().get_common_objects()
+        deviation_list = []
+        for e, s in self.request.session['already_have_deviation']:
+            deviation_list += [DeadlineRuleDeviation.objects.get(
```

Comment:
I think `DeadlineRuleDeviation.objects.get` raises an exception if the object does not exist in the database anymore. You must handle the exception. It is possible that users delete rows at the same time somebody tries to add more rows.

Is it also unnecessary to use the `+=` operator of lists here? You can `append` single elements to the list. That would be a better implementation here, right?